pkgs/shells/zsh/fzf-zsh/default.nix

```diff
@@ -23,8 +23,11 @@ stdenv.mkDerivation rec {
 
   dontBuild = true;
 
+  outputs = [ "out" "interactiveShellInit_zsh"];
+
   installPhase = ''
     install -Dm0644 fzf-zsh.plugin.zsh $out/share/zsh/plugins/fzf-zsh/fzf-zsh.plugin.zsh
+    install -Dm0644 fzf-zsh.plugin.zsh $interactiveShellInit_zsh
```

Comment:
I think it would be a better design to only have the interactiveShellInit output and subdirectories for bash, fish and zsh. This would also be easier to extend.

Reply:
I've chosen this design for a couple of reason:

1. It allow users of, say, zsh to not pay for fish plugins
2. It allows `programs,zsh,shellPkgs` to contain only packages with zsh plugins
3. In the immediate future I intend to create (either here or on home-manager) a module that allows me to manage zsh plugins to be loaded and managed by zplugin to improve parsing even further.
4. Especially point 3 require IFD with your design

In short I need to filter plugins by shell without IFD and I can't do that with per shell subdirs

Reply:
As for extensibility, you just have to add to the lists in the imports in https://github.com/NixOS/nixpkgs/blob/a4e97999decd694aba59773ae384d030a97e08e0/nixos/modules/config/shells-pkgs.nix#L71